### PR TITLE
AWS API Gateway Module Code Modification

### PR DESCRIPTION
**Pull request by kedarghule**
API Gateway Code modification done as per the KMS PR. to resolve issue - https://github.com/Cloudanix/sdlc/issues/169

Do review it, merge it and then test it comprehensively.

Thanks :)

### CODE REVIEW

File: cartography/data/indexes.cypher [hunk 1]
```diff
@@ -112,7 +115,8 @@ CREATE INDEX ON :RedshiftCluster(arn);
 CREATE INDEX ON :RDSInstance(db_instance_identifier);
 CREATE INDEX ON :RDSInstance(id);
 CREATE INDEX ON :RDSInstance(arn);
-CREATE INDEX ON :ReplyUri(id)
+CREATE INDEX ON :ReplyUri(id);
```
Comment (mpurusottamc) [inline cartography/data/indexes.cypher:118]: Was this deleted by us by mistake?

File: cartography/data/indexes.cypher [hunk 2]
```diff
@@ -112,7 +115,8 @@ CREATE INDEX ON :RedshiftCluster(arn);
...
+CREATE INDEX ON :RestAPI(id);
```
Comment (mpurusottamc) [inline cartography/data/indexes.cypher:119]: Let's rename it to APIGatewayRestAPI for clarity

File: cartography/intel/aws/apigateway.py [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+@timeit
+@aws_handle_regions
+def get_apigateway_rest_apis(boto3_session, region):
+    client = boto3_session.client('apigateway', region_name=region)
+    paginator = client.get_paginator('get_rest_apis')
+    apis = []
+    for page in paginator.paginate():
+        apis.extend(page['items'])
+    return apis
```
Comment (mpurusottamc) [inline cartography/intel/aws/apigateway.py:22]: Do you think we need to do as_dict() similar to what we did for KMS?

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
Minor changes needed. Also, please update docs/schema/awsmd with API Gateway resource details.

File: cartography/data/indexes.cypher (hunk 1 above)
Comment (mpurusottamc) [reply cartography/data/indexes.cypher:118]: never mind, i see the difference (;)

File: cartography/intel/aws/apigateway.py (hunk 3 above)
Comment (kedarghule) [reply cartography/intel/aws/apigateway.py:22]: We didn't use as_dict() in KMS. That was the Azure codes

File: cartography/data/indexes.cypher (hunk 2 above)
Comment (kedarghule) [reply cartography/data/indexes.cypher:119]: done

**Review by mpurusottamc — APPROVED** [review]
Code looks good.

**Pull request merged**